src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -37,6 +37,12 @@ private Rewriter(bool isAllThrowStatements)
                 bool shouldMoveNextStatementToSwitchExpression,
                 bool generateDeclaration)
             {
+                if (switchStatement.ContainsDirectives)
+                {
+                    // Do not rewrite statements with preprocessor directives
+                    return switchStatement;
+                }
```

Comment:
How can this condition be hit? The code fix bails-out early, so the rewriter wouldn't be called.

https://github.com/dotnet/roslyn/blob/6e549320f8c9b13ed732fa40a1f533aee0593f97/src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.cs#L49-L53

Reply:
I added `TestMissingFixAllOnDirectives` specifically for this case. The custom Fix All provider can be invoked from any supported location in the document, and gathers up _all_ diagnostics for the document (including ones that cannot be automatically fixed).